packages/amplify-category-api/src/provider-utils/awscloudformation/cfn-api-artifact-handler.ts

```diff
@@ -260,6 +260,7 @@ class CfnApiArtifactHandler implements ApiArtifactHandler {
  */
 export const writeResolverConfig = async (conflictResolution: ConflictResolution, resourceDir) => {
   const localTransformerConfig = await readTransformerConfiguration(resourceDir);
-  localTransformerConfig.ResolverConfig = conflictResolutionToResolverConfig(conflictResolution);
+  const resolverConfig = conflictResolutionToResolverConfig(conflictResolution);
+  if (!_.isEmpty(resolverConfig)) localTransformerConfig.ResolverConfig = resolverConfig;
```

Comment:
code lgtm 
can we also update the amplify e2e tests so to check that the resolver config does not exist when adding a non datastore api?